src/services/gateway.ts

```diff
@@ -1,32 +1,59 @@
 import type { Paginated } from "./pagination";
 
+export interface PaginatedResult<T> {
+  next: string | null;
+  previous: string | null;
+  page: T[];
+}
+
+//Dit is aangepast in de context van OpenZaak.
+//Het is nog onduidelijk of een generieke pagination functie bruikbaar blijft.
+//De data komt namelijk niet langer een enkele gateway.
+//Mogelijk refactoren naar api specifieke paginerings functies.
+//oude situatie hebben we ter referentie nog even uitgecommentarieerd laten staan
+
 export async function parsePagination<T>(
   json: unknown,
   map: (jObj: unknown) => T
-): Promise<Paginated<Awaited<T>>> {
-  const { results, limit, total, page, pages } = json as {
+): Promise<PaginatedResult<Awaited<T>>> {
+  const { count, next, previous, results } = json as {
     [key: string]: unknown;
   };
   if (
     !Array.isArray(results) ||
-    typeof limit !== "number" ||
-    typeof total !== "number" ||
-    typeof page !== "number" ||
-    typeof pages !== "number"
+    typeof count !== "number" ||
```

Comment:
Als we er achter kunnen komen hoe groot de page size is (en of je die kan instellen), kan je volgens mij de overige getallen berekenen.

Reply:
Goed punt. Had ik niet aan gedacht.